sirepo/package_data/static/js/raydata.js

```diff
@@ -30,70 +33,88 @@ SIREPO.app.factory('raydataService', function(appState, requestSender, simulatio
 	return cols.length > 0 ? [firstColHeading].concat(cols) : [];
     };
 
-    self.getScansInfo = function(scanUuids, successCallback, options) {
-	function maybeGetOnlyOneScan(scans) {
-	    return s ? scans[0] : scans;
-	}
+    self.getScansInfo = function(successCallback, options) {
+	function helper(successcallback, options, recursionDepth) {
+	    const s = Object.keys(appState.models.scans.selected);
+	    if (s.every((e) => {
+		// POSIT: If start is present so are the other fields we need
+		return e in (simulationDataCache.scans || {}) && simulationDataCache.scans[e].start;
```

Comment:
I think a utility could work.

Creating something like PKDict seems problematic because we have so much code that wouldn't be using it and from just looking at code I don't think you would be able to know whether the code was a plain JS object or one of our new objects. I think that would lead to bugs. Am I right to be concerned about this?

Here is how this code would possibly look using the utility:
```js
function jkunchecked_nested_get(obj, dottedKey) {
    let o = obj;
    for (const k of dottedKey.split('.')) {
        if (o === undefined) {
            return null;
        }
        o = o[k];
    }
    return o;
}


return jkunchecked_nested_get(simulationDataCache, `scans.${e}.start`)
```

I think that is a big improvement.

Thoughts?

Reply:
I agree this is a big improvement. I don't like the name (in pkcollections either :). The "jk" sounds like just kidding. I think this is a sirepo specific so something like "srxxx". 

Do we have functions that just float around? Perhaps we need something like sr.nestedGet where `sr` is a global with utility functions. Perhaps it could also be a factory. Maybe something relating to classes. I think the argument could also be a list, e.g. sr.nestedGet(simulationDataCache, ['scans', e, 'start']). It's a bit ugly, but it would work.

I think an object might be better. I think it wouldn't lead to bugs, because calling a function (nestedGet) will fail with an exception if the Object's attribute doesn't exist or isn't a function. 

@moellep?